### DISCUSSION TITLE
About empty handler

### DISCUSSION THREAD

**Discussion by josejachuf** [salvo-rs/salvo#220] [Q&A] [answered]
Hi
What is empty handler used? 
What is the reason for its existence?

`
let router = Router::with_hoop(cors_handler).get(hello).options(empty_handler);`

**Comment by chrislearn** [comment, marked as answer]
We only want `cors_handler` executed but don't want do anything else. If we do not add `.options(empty_handler)`, this router match will failed and returns 404 error.

**Comment by josejachuf** [comment]
My route wheel is like that and works without .options(empty_handler)

```
Router::new().hoop(cors_handler)
        .get(health::get_health)
```

**Reply by chrislearn** [reply]
Your router works with `GET` method, but if request with `OPTIONS` method, it will returns 404.

**Reply by josejachuf** [reply]
Thanks!

**Comment by josejachuf** [comment]
why always I get this warning?

`2023-01-13T17:35:42.967811Z  WARN salvo_core::service: Http response content type header not set uri=/api/v1/some method="OPTIONS"
`


My cors handles is:

```
let cors_handler = Cors::builder()
        .allow_any_origin()
        // .allow_origin("http://127.0.0.1:8000")
        .allow_methods(vec!["GET", "POST", "OPTIONS"])
        .allow_headers(vec![
            "CONTENT-TYPE",
            "content-type",
            "Access-Control-Request-Method",
            "Access-Control-Allow-Origin",
            "Access-Control-Allow-Headers",
            "Access-Control-Max-Age",
            "authorization",
            "Authorization",
        ])
        .build();
```

**Reply by chrislearn** [reply]
I will fix this, avoid give this warning when method is `OPTIONS`.

**Reply by josejachuf** [reply]
Thanks!
